src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -3801,14 +3803,14 @@ public static void Main()
             }
             """, "Program.cs");
 
-        var interceptor = ("""
+        var interceptor = ($$"""
             using System.Runtime.CompilerServices;
             using System;
 
             static class D
             {
                 [InterceptsLocation("Program.cs", 9, 23)]
-                public static void Interceptor(this in Program x) => Console.Write("Intercepted");
+                public static void Interceptor(this {{refKind}} Program x) => Console.Write("Intercepted");
```

Comment:
Do we have a similar test where the intereptor has a `ref readonly` parameter (but it's not the receiver) and the callsite passes an rvalue to it? And similarly when an lvalue is passed but without any `ref`/`in` callsite modifier. Both would be warnings if those were non-intercepted invocations. I'm wondering if there should be a warning for the intercepted call as well (the interceptor can just use `in` to avoid the warning)?

Reply:
I believe the answer here is the same as for https://github.com/dotnet/roslyn/pull/74363#discussion_r1677358306, i.e. only the original method parameters are used to decide to report a warning about a wrong argument value kind or callsite modifier.